In [7]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt


In [2]:
df=pd.read_csv('/home/athira/Desktop/dsa/StudentsPerformance.csv')

In [8]:
df.head()

gender race/ethnicity parental level of education         lunch  \
0  female        group C            some high school  free/reduced   
1  female        group B                 high school  free/reduced   
2  female        group B            some high school  free/reduced   
3  female        group B                some college      standard   
4  female        group C                some college  free/reduced   

  test preparation course  math score  reading score  writing score  
0                    none           0             17             10  
1                    none           8             24             23  
2                    none          18             32             28  
3                    none          11             38             32  
4                    none          22             39             33

1.find out how manay males and females participants in the test

In [9]:
gender=df['gender'].value_counts()
gender


female    518
male      482
Name: gender, dtype: int64

2.What do you think about the students parantel level of education?

In [12]:
df['parental level of education'].value_counts()


some college          226
associate's degree    222
high school           196
some high school      179
bachelor's degree     118
master's degree        59
Name: parental level of education, dtype: int64

226 parents have college education. 2. out of 226, 118 have bachelors degree and 59 have masters degree

3.Who scores the most on average for math, reading and writing based on
● Gender


In [13]:
df1 = df[['gender', 'math score', 'reading score', 'writing score']].groupby('gender').mean()
df1

math score  reading score  writing score
gender                                          
female   63.376448      72.590734      72.467181
male     68.821577      65.545643      63.446058

Female students have higher average reading and writing score while male students have higher average in maths score

● Test preparation course

In [14]:
df2= df[['test preparation course', 'math score', 'reading score', 'writing score']].groupby('test preparation course').mean()
df2

math score  reading score  writing score
test preparation course                                          
completed                  69.96648      74.175978      74.684358
none                       63.78972      66.417445      64.457944

Students who did test preparation course have higher average score across all categories.

4.What do you think about the scoring variation for math, reading and writing
based on
● Gender
● Test preparation course

In [16]:
genvar= df[['gender', 'math score', 'reading score', 'writing score']].groupby('gender').std()
genvar


math score  reading score  writing score
gender                                          
female   16.029928      14.411018      14.844842
male     14.556411      14.149594      14.227225

In [17]:
testvar= df[['test preparation course', 'math score', 'reading score', 'writing score']].groupby('test preparation course').std()
testvar

math score  reading score  writing score
test preparation course                                          
completed                 14.521847      13.537572      13.236412
none                      15.705689      14.608896      15.041667

male students have less variations across each of the scores(14.556411 ,14.149594 ,14.227225).
The variation between those who completed and none are more or less the same.


5.The management needs your help to give bonus points to the top 25% of
students based on their math score, so how will you help the management
to achieve this.

In [19]:
df.nlargest(250, ['math score'])


gender race/ethnicity parental level of education         lunch  \
515  female        group E                some college      standard   
516  female        group E           bachelor's degree      standard   
517  female        group E          associate's degree      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
..      ...            ...                         ...           ...   
428  female        group D                some college  free/reduced   
429  female        group E            some high school      standard   
849    male        group A           bachelor's degree      standard   
850    male        group C          associate's degree  free/reduced   
851    male        group D                some college      standard   

    test preparation course  math score  reading score  writing score  
515                    none         100             92             97  
516                    none         100            100            100  
517                    none         100            100            100  
996               completed         100             96             86  
997               completed         100             97             99  
..                      ...         ...            ...            ...  
428                    none          77             86             86  
429                    none          77             79             80  
849                    none          77             67             68  
850                    none          77             67             64  
851               completed          77             62             62  

[250 rows x 8 columns]

List of top 250 students who are eligible for bonus points